python/mp/effects/bin.py

```diff
@@ -0,0 +1,99 @@
+import copy
+from mp import color
+from mp.effects import effect
+
+class Bin(effect.Effect):
+    """
+    This effect contains a list of other effects. Calling next() invokes the next() method
+    on all the contained effects. Rosary actually only contains this one effect, adding and
+    removing effects to and from it.
+
+    In other words, its turtles all the way down.
+    """
+
+    # Wish there were a better way than requiring this every time
+    dm = copy.deepcopy(effect.Effect.dm)
+
+    def __init__(self, bead_set, **kwargs):
+        super().__init__(name="bin", bead_set=bead_set, **kwargs)
+        self.effects = []
+        self.effect_id_num = 0
+
+    def effect(self, id):
+        """Return the Effect object of an active effect by specifying the Effect id."""
+        for e in self.effects:
+            if e.id == id:
+                return e
+        return None    
+
+    def effect_id(self):
+        self.effect_id_num += 1
+        return self.effect_id_num
+
+    def add_effect_object(self, effect):
+        """Adds an Effect object to the active Effect list.  Returns the id of
+        the active effect.
+
+x        """
+        effect.id = self.effect_id()
+        # Since rosary holds the dispatcher and the effect doesn't
+        # know about rosary on init, we can't map to dispatcher yet either
+        self.effects.append(effect)
+        effect.rosary = self.rosary
+
```

Comment:
That comment was from before I refactored the OSC path mapping stuff, when the effect had to have a reference to the rosary to do it. I could've had rosary do it from the outside, but that was icky. You're right though, that's no longer true.